includes/wikia/parser/templatetypes/tests/InfoiconTemplateTest.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+class InfoiconTemplateTest extends WikiaBaseTest {
+	/**
+	 * @param $infoiconWikitext
+	 * @param $expectedTemplateWikiext
+	 *
+	 * @dataProvider testHandleDataProvider
+	 */
+	public function testHandle( $infoiconWikitext, $expectedTemplateWikiext ) {
+		$sanitizedTemplateWikiext = InfoiconTemplate::handle( $infoiconWikitext );
+
+		$this->assertEquals( $sanitizedTemplateWikiext, $expectedTemplateWikiext );
+	}
+
+	public function testHandleDataProvider() {
+		return [
+			[
+				'[[:Harry Potter]]',
+				'[[:Harry Potter]]'
```

Comment:
How this will render if source image has higher base dimensions than the limit for ICON on Mercury?